src/pages/PrimeCalculator/Form/index.tsx

```diff
@@ -112,13 +180,94 @@ export const Form: React.FC = () => {
     };
 
     fn();
-  }, [options, updateUrlTokenAddress, setValue, urlTokenAddress, defaultTokenAddress]);
+  }, [options, updateUrlTokenAddress, urlTokenAddress, defaultTokenAddress]);
+
+  const { control, formState, setValue, reset, watch } = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      stakedAmountXvsTokens: '',
+      suppliedAmountTokens: '',
+      borrowedAmountTokens: '',
+    },
+  });
+
+  // Load default form values
+  const { initialStakedAmountXvsTokens, initialBorrowAmountTokens, initialSupplyAmountTokens } =
+    useMemo(() => {
+      let stakedAmountXvsTokens;
+      let borrowAmountTokens;
+      let supplyAmountTokens;
+
+      if (selectedAsset && xvs) {
+        const defaultValues = getInitialValues({
+          assetData: selectedAsset,
+          primeMinimumStakedXvsMantissa: primeMinimumStakedXvsMantissa || new BigNumber('0'),
+          userXvsStakedMantissa,
+          xvs,
+        });
+        stakedAmountXvsTokens = defaultValues.initialStakedAmountXvsTokens;
+        borrowAmountTokens = defaultValues.initialBorrowAmountTokens;
+        supplyAmountTokens = defaultValues.initialSupplyAmountTokens;
+      }
+      return {
+        initialStakedAmountXvsTokens: stakedAmountXvsTokens,
+        initialBorrowAmountTokens: borrowAmountTokens,
+        initialSupplyAmountTokens: supplyAmountTokens,
+      };
+    }, [selectedAsset, xvs, primeMinimumStakedXvsMantissa, userXvsStakedMantissa]);
+
+  useEffect(() => {
+    const stakedAmountXvsTokens = initialStakedAmountXvsTokens
+      ? initialStakedAmountXvsTokens.toFixed()
+      : '';
+    const borrowAmountTokens = initialBorrowAmountTokens ? initialBorrowAmountTokens.toFixed() : '';
+    const supplyAmountTokens = initialSupplyAmountTokens ? initialSupplyAmountTokens.toFixed() : '';
+    if (formState.touchedFields.stakedAmountXvsTokens) {
+      setValue('borrowedAmountTokens', borrowAmountTokens);
+      setValue('suppliedAmountTokens', supplyAmountTokens);
+    } else {
+      reset({
+        stakedAmountXvsTokens,
+        suppliedAmountTokens: supplyAmountTokens,
+        borrowedAmountTokens: borrowAmountTokens,
+      });
+    }
+  }, [
+    initialStakedAmountXvsTokens,
+    initialBorrowAmountTokens,
+    initialSupplyAmountTokens,
+    formState.touchedFields.stakedAmountXvsTokens,
+    setValue,
+    reset,
+  ]);
+
+  const { borrowedAmountTokens, stakedAmountXvsTokens, suppliedAmountTokens } = watch();
```

Comment:
I would put in the form santizing inputs and returning them in the correct unit to be consumed by the app. This will clean up downstream logic where these values get converted to different units